### PR TITLE
Allow factory to contain projects with keepDir function

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Projects that are created with keepDir option have no source type or location.
So the source validation or project in factory will be omitted, if such project is located in another project

### CODE REVIEW

**Comment by benoitf** [comment]
it seems commit is missing related issue in the message

**Comment by mkuznyetsov** [comment]
will fix that during merge

File: wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectManager.java [hunk 1]
```diff
@@ -333,6 +333,12 @@ public RegisteredProject importProject(String path, SourceStorage sourceStorage,
         List<? extends ProjectConfig> projects = workspaceConfig.getProjects();
         for (ProjectConfig project : projects) {
             if (normalizePath.equals(project.getPath())) {
+                // TODO Needed for factory project importing with keepDir. It needs to find more appropriate solution
+                List<String> projects1 = projectRegistry.getProjects(normalizePath);
+                for (String s : projects1) {
```
Comment (skabashnyuk) [inline wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectManager.java]: projects1  - nasty name. Can you come up new one?

**Comment by skabashnyuk** [comment]
other ok

File: wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectManager.java (hunk 1 above)
Comment (sleshchenko) [inline wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectManager.java]: s -> project?

Comment (mkuznyetsov) [inline wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectManager.java]: fixed

**Pull request merged**